app/services/subscriptions/dates/quarterly_service.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+module Subscriptions
+  module Dates
+    class QuarterlyService < Subscriptions::DatesService
+      private
+
+      def compute_from_date(date = base_date)
+        if plan.pay_in_advance? || terminated_pay_in_arrear?
+          return subscription.anniversary? ? previous_anniversary_day(billing_date) : billing_date.beginning_of_quarter
+        end
+
+        subscription.anniversary? ? previous_anniversary_day(date) : date.beginning_of_quarter
+      end
+
+      def compute_charges_from_date
+        if terminated?
+          return subscription.anniversary? ? previous_anniversary_day(billing_date) : billing_date.beginning_of_quarter
+        end
+
+        return compute_from_date if plan.pay_in_arrear?
+        return base_date.beginning_of_quarter if calendar?
+
+        previous_anniversary_day(base_date)
+      end
+
+      def compute_charges_to_date
+        return compute_charges_from_date.end_of_quarter if calendar?
+
+        compute_to_date(compute_charges_from_date)
+      end
+
+      def compute_duration(from_date:)
+        next_to_date = compute_to_date(from_date)
+
+        (next_to_date.to_date + 1.day - from_date.to_date).to_i
+      end
+
+      alias compute_charges_duration compute_duration
+
+      def compute_base_date
+        billing_date - 3.months
+      end
+
+      def compute_to_date(from_date = compute_from_date)
+        if subscription.calendar? || (subscription_at.day == 1 && [1, 4, 7, 10].include?(subscription_at.month))
+          return from_date.end_of_quarter
+        end
+
+        year = from_date.year
+        month = from_date.month + 3
+        day = subscription_at.day - 1
+
+        if month > 12
+          month = (month % 12) == 0 ? 12 : (month % 12)
```

Comment:
Style/NumericPredicate: Use `(month % 12).zero?` instead of `(month % 12) == 0`. (https://rubystyle.guide#predicate-methods)